numpy/ma/tests/test_core.py

```diff
@@ -1808,6 +1808,11 @@ def test_default_fill_value_structured(self):
         assert_equal(f1, expected)
         assert_equal(f2, expected)
 
+    def test_default_fill_value_void(self):
+        dt = np.dtype([('v', 'V7')])
+        f = default_fill_value(dt)
```

Comment:
The current infinite recursions are annoying for testing...

Also, here's a related bug fixed by this PR:
```
np.ma.array([b''], dtype='V7', mask=[True]).filled()
array(['???'], dtype=object)
```